tests/flytekit/unit/core/test_flyte_file.py

```diff
@@ -52,6 +52,79 @@ def my_wf() -> FlyteFile[typing.TypeVar("txt")]:
+    # TXT
+    @task
+    def t1() -> FlyteFile[typing.TypeVar("txt")]:
+        fname = "/tmp/flytekit_test.txt"
+        with open(fname, "w") as fh:
+            fh.write("Hello World\n")
+        return fname
+
+    @workflow
+    def my_wf() -> FlyteFile[typing.TypeVar("txt")]:
+        f = t1()
+        return f
+
+    res = my_wf()
+    print(type(res))
+    with open(res, "r") as fh:
+        assert fh.read() == "Hello World\n"
+
+
+def test_mismatching_file_types():
+    @task
+    def t1() -> FlyteFile[typing.TypeVar("jpeg")]:
+        fname = "/tmp/flytekit_test.txt"
+        with open(fname, "w") as fh:
+            fh.write("Hello World\n")
+        return fname
+
+    @workflow
+    def my_wf() -> FlyteFile[typing.TypeVar("jpeg")]:
+        f = t1()
+        return f
+
+    with pytest.raises(TypeError) as excinfo:
+        my_wf()
+    assert "Incorrect type, expected image/jpeg, got text/plain" in str(excinfo.value)
+
+
+def test_get_mime_type_from_python_type_success():
+    transformer = TypeEngine.get_transformer(FlyteFile)
+    assert transformer.get_mime_type_from_python_type("html") == "text/html"
+    assert transformer.get_mime_type_from_python_type("jpeg") == "image/jpeg"
+    assert transformer.get_mime_type_from_python_type("png") == "image/png"
+    assert transformer.get_mime_type_from_python_type("hdf5") == "application/x-hdf"
+    assert transformer.get_mime_type_from_python_type("joblib") == "application/octet-stream"
+    assert transformer.get_mime_type_from_python_type("pdf") == "application/pdf"
+    assert transformer.get_mime_type_from_python_type("python_pickle") == "application/octet-stream"
+    assert transformer.get_mime_type_from_python_type("ipynb") == "application/x-ipynb+json"
+    assert transformer.get_mime_type_from_python_type("svg") == "image/svg+xml"
+    assert transformer.get_mime_type_from_python_type("csv") == "text/csv"
+    assert transformer.get_mime_type_from_python_type("onnx") == "application/octet-stream"
+    assert transformer.get_mime_type_from_python_type("tfrecord") == "application/octet-stream"
+    assert transformer.get_mime_type_from_python_type("txt") == "text/plain"
+
+
+def test_get_mime_type_from_python_type_failure():
+    transformer = TypeEngine.get_transformer(FlyteFile)
+    with pytest.raises(KeyError):
+        transformer.get_mime_type_from_python_type("unknown_extension")
+
+
+def test_validate_file_type_incorrect():
+    transformer = TypeEngine.get_transformer(FlyteFile)
+    source_path = "/tmp/flytekit_test.png"
+    source_file_mime_type = "image/png"
+    user_defined_format = "jpeg"
+
+    FlyteFilePathTransformer.get_format = MagicMock(return_value=user_defined_format)
+    with patch("magic.from_file", return_value=source_file_mime_type):
+        with pytest.raises(ValueError, match=f"Incorrect file type, expected image/jpeg, got {source_file_mime_type}"):
+            transformer.validate_file_type(user_defined_format, source_path)
+
+
```

Comment:
Can you add test where `FlyteFile` is `Annotated`?

Reply:
Sorry, I don't understand what you mean. Can you give me some more hints or suggestions?

Reply:
I've added here.
https://github.com/flyteorg/flytekit/blob/35ddf821a2985c26faf6c5116e387a44f091719d/tests/flytekit/unit/core/test_flyte_file.py#L159-L178